treeherder/webapp/api/hash.py

```diff
@@ -0,0 +1,28 @@
+import logging
+
+from rest_framework import exceptions, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.status import HTTP_400_BAD_REQUEST
+
+from treeherder.model.models import Commit
+from treeherder.webapp.api.serializers import HashQuerySerializer
+
+logger = logging.getLogger(__name__)
+
+
+class HashViewSet(viewsets.ViewSet):
+    @action(detail=False)
+    def tocommit(self, request, project):
+        query_params = HashQuerySerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors, status=HTTP_400_BAD_REQUEST)
+        newhash = query_params.validated_data["newhash"]
+        basehash = query_params.validated_data["basehash"]
+        newpush = Commit.objects.filter(comments__contains=newhash).first()
+        basepush = Commit.objects.filter(comments__contains=basehash).first()
+        if newpush is None or basepush is None:
+            raise exceptions.ValidationError(
+                f"{newhash} or {basehash} do not correspond to any existing hashes please double check both hashes you provided"
+            )
+        return Response({"originalRevision": basepush.revision, "newRevision": newpush.revision})
```

Comment:
Is there a reason you're using originalRevision here instead of baseRevision? I think using baseRevision would be better to keep naming consistent.

Reply:
Changed to be baseRevision now